### ISSUE TITLE
no sync

### ISSUE THREAD

**Issue by nheid** [gpodder/GpodRoid#3]
as reported by prof morbius:

gpodroid successfully logged into gpodder.net and created a device, but after I synced the list from another device ... nothing.  No subscriptions are listed; all I get is a player with nothing to play and three blank tabs (the search field shows up in the Search tab, but returns nothing no matter what I type).  The comments in the Market listed a similar problem.

**Comment by brianbourke75** [comment]
So I believe that I figured out what is causing the syncing issue.  It appears that the ObjectMapper "read" method is failing to parse the JSON feedback when the API call is made to fetch subscription updates.  When reviewing a manual call to gpodder.net it appears there are new values which are not represented in our class structures thus the ObjectMapper seems to be getting confused.  Personally I believe it may be worth a look to see if there are more robust ways of handling the data and I will do some research to figure this out.

In the mean time I am plagued by logistical issues.  I updated my android SDK and that seems to have broken my build.xml file so I am trying to figure out how to fix that.  Also, I have never been able to get the integrated debugging to work under eclipse and I was wondering if you have had any luck with this?

**Comment by nheid** [comment]
hey, i was working on it too. and got so far, that i added the missing field.
still the db operations failed. but i only looked very briefly due to lack of time.

of the sdk debug issue, lets meet up in irc next week, i got it running in emulator and on device (even though i'm using intellij).

**Comment by brianbourke75** [comment]
Hiya,  I finally carved out some time to work on the project and I was able to fix the database issue.  Nothing complicated, but I decided to change the GpodDB class to a static class and make some other small changes which seemed to do the trick, but there seems to be another JSONMappingException being throw now, but at least it's a _new_ issue.  So I noticed your branch so I merged that with my local master here so we don't get to far from each other.

I will let you know if I figure out what the new issue is should I be able to figure it out.

ps: I was able to figure out the integrated debugging thing, I was just being a bit bone headed.

**Comment by brianbourke75** [comment]
So I was able to figure out what was going on.  I happened to have the podcast "FLOSS Weekly" in my subscription list and it seems to have some bad mojo in it's description.  Because of this I cannot get rid of it from the web interface or gpodder on my desktop so  I am thinking that this is an issue which should be handled on the backend. 

Do you have a contact person who works in the API or should I just message someone myself?

**Comment by nheid** [comment]
hey,
i didnt have any time to look in the db, that is why there is a branch
at the moment. (for release 0.4.4)
it's not intended to stay that way. was just because i got a handful
of "no sync" complaints.
i still have two crash reports in the market logs i want to fix, which
are related to empty results and network calls not detached from the
ui. that should increase stability a bit.

you can use the irc of the gpodroid guys, they are both very nice and helpful.
i had irc conversations with both of them, also about issues. they
usually took care of it within a few hours.
but if you want i can also relay the information. whatever you prefer.

i'll take a look into the parsing next week to check if i can make it
less sensible to changes in the api.

i'll be off for the weekend, so i wish you a nice weekend.

greetings, nico

On Fri, Sep 2, 2011 at 19:42, brianbourke75
reply@reply.github.com
wrote:

> So I was able to figure out what was going on.  I happened to have the podcast "FLOSS Weekly" in my subscription list and it seems to have some bad mojo in it's description.  Because of this I cannot get rid of it from the web interface or gpodder on my desktop so  I am thinking that this is an issue which should be handled on the backend.
> 
> Do you have a contact person who works in the API or should I just message someone myself?
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/gpodder/GpodRoid/issues/3#issuecomment-1981370

## 

Nico Heid - http://nicoheid.com - mob. +491733181059

**Comment by Horrendus** [comment]
Hey,

gpodder.net introduces new keys in the JSON results of the diverse API Calls from time to time (on request).
Please change your JSON -> Object Mappers in a way to only fail if there are missing keys, if they atm also fail if there are additional values

**Comment by brianbourke75** [comment]
Yup, I have been wanting to write a better JSON parser for a while now so now all I have to do is overcome that darn learning curve!

**Comment by Horrendus** [comment]
Hmm ... just found this: http://wiki.fasterxml.com/ObjectMapper

So they set some Property to not fail on an Unknown Value, maybe try this (Jackson seems not to be the best documented java lib)

**Comment by gpbenton** [comment]
I seemed to be having a similar problem, but the list of podcasts appeared after I pressed the `play´ button.  It put out a message like ´you must select a podcast first´, and then displayed all the synched podcasts.

**Comment by dalb8** [comment]
no such luck here.

**Comment by dalb8** [comment]
oh yes, patience paid off. I stopped it an d started it and 30 minutes later it works. Not ready for prime time, guess I'll have to keep using beyondpod for another while:(

**Comment by nheid** [comment]
hey,
yeah sorry for that, i took the app off the market a while ago.
due to this issue and my lack of time i think that was the only choice.

thanks for your feedback and support.

sorry again,
nico

On Mon, Jun 4, 2012 at 10:29 PM, dalb8 <
reply@reply.github.com

> wrote:
> 
> oh yes, patience paid off. I stopped it an d started it and 30 minutes
> later it works. Not ready for prime time, guess I'll have to keep using
> beyondpod for another while:(
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/gpodder/GpodRoid/issues/3#issuecomment-6110125

## 

Nico Heid - http://nicoheid.com - mob. +491733181059
